lib/scope.js

```diff
@@ -1125,28 +1113,87 @@ module.exports = {
     // TODO: Maybe we should return something else or something in addition about used vars
   },  // Ends scope.processVar()
 
-  getThereIsAnotherValue: async function ( scope, { value, from_table=true }) {
-    /* Given story row-like data for a `.there_is_another` var, as well as
-    * whether this is from a table row, return the best guess as to a
-    * safe value to set for this round. Give the developer a warning if
+  ensureSpecialRows: async function ( scope, { var_data, from_table=true }) {
+    /* Given a list of variable data objects, add more objects or mutate
+    * rows under special circumstances:
+    * - `.target_number` may indicate the need for a `.there_is_another` row.
+    * - `.there_is_another` in a table is invalid
+    */
+    // Add special cases. Do not mutate `var_data` as a list, but
+    // do allow mutation of items inside.
+    let enhanced_var_data = [...var_data];
+    for ( let var_datum of var_data ) {
+
+      // Add a `.there_is_another` var if it might be needed.
+      // If it's not used in the end, that's fine.
+      // `.target_number` is sometimes a stand-in for `.there_is_another`.
+      // That seems the clearest way to communicate to developers about what they
+      // should write in the test. We will attempt to write more about
+      // the rationale in the documentation at some point.
+      if ( from_table && var_datum.var.match( /\.target_number$/ )) {
+        let special_row = {
+          artificial: true,
```

Comment:
We aren't using this `artificial` value anywhere right now, that I can tell.

Reply:
That's true. It makes me nervous to have a non-dev generated row around without explicitly saying it. It's a bit of premature future-proofing. Is it worth keeping around or is it confusing?

Reply:
In my head, `source` performs the same function, as it only exists for non-dev generated rows. Can we just check for the existence of `row.source`?

Reply:
I'm fine with keeping this around too, but maybe with a comment saying it is not currently used but is future-proofing to mark non-dev generated rows.